### DISCUSSION TITLE
SwiftUI Snapshot doesn't match simulator

### DISCUSSION THREAD

**Discussion by humblehacker** [pointfreeco/swift-snapshot-testing#595] [Q&A]
I'm testing out snapshot testing in a prototype app, and I've noticed that the snapshots taken don't match an actual simulator screenshot, at least for Form fields. For example, notice how in the snapshot image each text field in the form seems to be in its own bubble.

| Simulator | Snapshot |
|---|---|
| <img alt="RocketSim_Screenshot_iPhone_Xs_Max_2022-04-06_12 10 39" src="https://user-images.githubusercontent.com/117582/162051044-c8e8aba3-d00a-41f1-a0bb-5bd78695c0f1.png" width=300> | <img alt="testExample 1" src="https://user-images.githubusercontent.com/117582/162051076-3a00feb5-362a-4394-b38f-be11b7dcf4ba.png"  width=300> |

Can anyone account for the differences? Should I expect other views to have similar discrepancies? Or is there something I can do to correct it myself? Sorry if this has been answered already. I couldn't find anything when I searched.

**Comment by heckj** [comment]
Part of the difference looks to be "the view minus a navigation controller", and SwiftUI in particular changes how it renders things based on context - so it wouldn't surprise me if that were part of the scenario here.
